### ISSUE TITLE
Error parsing mongodb multi-host url (ValueError)

### ISSUE THREAD

**Issue by rodolfo3** [celery/kombu#375]
I was using multiple mongodb hosts on celery configuration, but it is not working after a kombu update:

It works at 3.0.16:

```
>>> import kombu
>>> kombu.__version__
'3.0.16'
>>> from kombu.utils.url import _parse_url
>>> _parse_url("mongodb://localhost:27017,localhost:27018/celery")
('mongodb', 'localhost:27017,localhost:27018/celery', None, None, None, 'celery', {})
```

But breaks at 3.0.17:

```
>>> import kombu
>>> kombu.__version__
'3.0.17'
>>> from kombu.utils.url import _parse_url
>>> _parse_url("mongodb://localhost:27017,localhost:27018/celery")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rodolfo/.virtualenvs/learn/local/lib/python2.7/site-packages/kombu/utils/url.py", line 24, in _parse_url
    return (scheme, unquote(parts.hostname or '') or None, parts.port,
  File "/usr/lib/python2.7/urlparse.py", line 113, in port
    port = int(port, 10)
ValueError: invalid literal for int() with base 10: '27017,localhost'
```

**Comment by GoingTharn** [comment]
I still see this as an issue with 3.0.24 build. In fact, I haven't seen this work since 3.0.15

**Comment by daevaorn** [comment]
This patch works ok but only lands on master and never has been backported to 3.1.x branch.

**Comment by olivierlefloch** [comment]
Could this get back ported into the Celery 3.1 branch please?

**Comment by ask** [comment]
Merged into celery 3.1
